services/Skype.filters.js

```diff
@@ -0,0 +1,95 @@
+        "mainonedrivemodule",
+        "mainoutlookmodule",
+        "mainlinkedinmodule",
+        "mainsearchaimodule",
+        "mainbingmodule",
+        "maincortanamodule",
+        "mainmsedgemodule",
+        "mainmsedgemodule",
+        "mainMicrosoftTranslatormodule",
+        "mainswiftkeymodule",
+        "mainwindowsmodule",
+        "mainactivationmodule",
+        "mainactivityhistorymodule",
+        "mainadvertisingidmodule",
+        "maindiagnosticsmodule",
+        "mainfeedbackhubmodule",
+        "mainlocationservicesmotionsensingmodule",
+        "mainsecurityandsafetyfeaturesmodule",
+        "mainspeechinkingtypingmodule",
+        "mainsyncsettingsmodule",
+        "mainupdateservicesmodule",
+        "mainwebbrowsersmodule",
+        "mainwindowsappsmodule",
+        "mainwindowsmediaplayermodule",
+        "mainwindowshellomodule",
+        "mainwindowssearchmodule",
+        "mainyourphonemodule",
+        "mainentertainmentmodule",
+        "mainxboxmodule",
+        "mainwindowsstoremodule",
+        "mainmainmodule",
+        "mainmixermodule",
+        "maingroovemusicmoviestvmodule",
+        "mainsilverlightmodule",
+        "mainwindowsmixedrealitymodule"
+    ];
+    moduleDescriptions.forEach(moduleDescription => {
+        let moduleName = moduleDescription.querySelector('#moduleName');
+        if (moduleName) {
+            moduleName = moduleName.textContent;
+        }
+        if (moduleName && toDelete.includes(moduleName)) {
+            moduleDescription.remove();
+        }
+    });
+}
+
+export function removeIrrelevantModulesParentTos(document) {
+    const moduleDescriptions = document.querySelectorAll('.divModuleDescription');
+    moduleDescriptions.forEach(moduleDescription => {
+        var name = moduleDescription.querySelector('.FileName');
+        if (name) {
+            name = name.textContent;
+        }
+        if (name.substr(0,2)=="14" && name.substr(0,3)!="14e" && name.substr(0,3)!="14_") {
+            moduleDescription.remove();
+        }
+        if (name=="serviceslist") {
+            moduleDescription.remove();
+        }
+    });
+}
+
+export function removeNavigationHeaders(document) {
+    document.querySelectorAll('#navigationHeader, .navigationHeader').forEach(element => element.remove());
+}
+
+export function removeModuleIDs(document) {
+    document.querySelectorAll('#moduleName, .FileName').forEach(element => element.remove());
+}
+
+export function removeInvisibleText(document) {
+    const notDisplayed = document.querySelectorAll('.displayNone');
+    const summaries = document.querySelectorAll('.printsummary');
+    const fulltexts = document.querySelectorAll('.printDetailedContent');
+    notDisplayed.forEach(element => element.remove());
+    summaries.forEach(summary =>  {
+        if (summary.textContent=="Summary") {
+            summary.remove();
+        }
+    });
+    fulltexts.forEach(fulltext => {
+        if (fulltext.textContent=="Full text") {
```

Comment:
Why do we add these checks, both for Summary and Full text? When I check, I see no case where the class is set and the text content is different 🤔

Reply:
I thought it would be more robust, just in case, but tell me if this is irrelevant :)

Reply:
Thanks for clarifying! It is impossible to predict what “robustness” looks like until we've seen which evolutions are usual. Given the current lack of information, this condition makes the reader assume that there was good reason for checking the content, and this currently useless code will stay in place and possibly be maintained for no reason. Always strive for the simplest piece of code that can achieve the result 🙂

I've refactored the code already, and I'll let you remove this condition just so you can get a feel of how the map/reduce approach improves the maintainability of the code compared to the imperative approach.

Reply:
Done thanks for your useful advices ! :)